generators/base/generator-base.mjs

```diff
@@ -2500,6 +2505,77 @@ templates: ${JSON.stringify(existingTemplates, null, 2)}`;
       (dest.applicationType === MICROSERVICE && !dest.skipUserManagement);
 
     dest.generateBuiltInAuthorityEntity = dest.generateBuiltInUserEntity && !dest.databaseTypeCassandra;
+
+    if (dest.databaseTypeSql) {
```

Comment:
may be staring to stop using `databaseTypeSql` in favor of `isSql(devDb, prodDb)` helper instead? or making databaseTypeSql a derived property instead of a user-specified one.

Reply:
IMO `databaseTypeSql` should be kept or a alternative.

> or making databaseTypeSql a derived property instead of a user-specified one.

I don't get this remark.

Reply:
https://github.com/jhipster/generator-jhipster/issues/20679